core/src/main/java/io/opencensus/tags/TagUtils.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.tags;
+
+import io.opencensus.common.Function;
+
+final class TagUtils {
+  private TagUtils() {}
+
+  private static final Function<Object, Void> THROW_ASSERTION_ERROR =
+      new Function<Object, Void>() {
+        @Override
+        public Void apply(Object ignored) {
+          throw new AssertionError();
+        }
+      };
+
+  /**
+   * A {@code Function} that always ignores its argument and throws an {@link AssertionError}.
+   *
+   * @return a {@code Function} that always ignores its argument and throws an {@link
+   *     AssertionError}.
+   */
+  static <T> Function<Object, T> throwAssertionError() {
```

Comment:
Hiding this method sounds good to me, since I'm not quite sure about the use case either. Maybe having `throwIllegalArgumentException` and `throwAssertionError` together at `Functions` will cause confusion at user side, since it will be unclear which one is preferable for the default/undesired match case.